src/managers/GuildMemberManager.js

```diff
@@ -135,6 +136,50 @@ class GuildMemberManager extends BaseManager {
     return this._fetchMany(options);
   }
 
+  /**
+   * Edits a member of the guild.
+   * <info>The user must be a member of the guild</info>
+   * @param {UserResolvable} user The member to edit
+   * @param {GuildMemberEditData} data The data to edit the member with
+   * @param {string} [reason] Reason for editing this user
+   * @returns {Promise<void>}
+   */
+  async edit(user, data, reason) {
+    const id = this.client.users.resolveID(user);
+    if (!id) throw new TypeError('INVALID_TYPE', 'user', 'UserResolvable');
+
+    // Clone the data object for immutability
+    const _data = { ...data };
+    if (_data.channel) {
+      _data.channel = this.guild.channels.resolve(_data.channel);
+      if (!_data.channel || _data.channel.type !== 'voice') {
+        throw new Error('GUILD_VOICE_CHANNEL_RESOLVE');
+      }
+      _data.channel_id = _data.channel.id;
+      _data.channel = undefined;
+    } else if (data.channel === null) {
+      _data.channel_id = null;
+      _data.channel = undefined;
+    }
+    if (_data.roles) _data.roles = _data.roles.map(role => (role instanceof Role ? role.id : role));
+    let endpoint = this.client.api.guilds(this.guild.id);
+    if (id === this.client.user.id) {
+      const keys = Object.keys(_data);
+      if (keys.length === 1 && keys[0] === 'nick') endpoint = endpoint.members('@me').nick;
+      else endpoint = endpoint.members(id);
+    } else {
+      endpoint = endpoint.members(id);
+    }
+    await endpoint.patch({ data: _data, reason });
+
+    if (this.cache.has(id)) {
+      const member = this.cache.get(id);
+      const clone = member._clone();
+      _data.user = member.user;
+      clone._patch(_data);
```

Comment:
You..want to return the clone too... Otherwise this code block is useless

Reply:
This is one of those areas which I revised on the MessageManager PR - debating whether these methods should *sometimes* return an object if one was cached, and return raw API data or null in other cases.

Thoughts?

Reply:
It would be a breaking change to return raw api data... That, and in this endpoints case we get no data apparently? Soo..yeahh idk, it's up to discussion

Reply:
This is a brand new method so it can return whatever we like I would think?

Reply:
Yes, however you use the method in GuildMember which would change return types and make it a breaking PR, unless the aforementioned method returns the previous data

Reply:
The `GuildMember#edit` method doesn't rely on the return value of this - it just assumes that the Manager will have updated the cache if it's available, then returns `this`.

https://github.com/discordjs/discord.js/blob/b0c8f740fd97bb0550d4cf623b09748b519861ce/src/structures/GuildMember.js#L292

Reply:
This method won't update the cache however, only the clone, which is currently not being returned.

It however might be worth a consideration to actually update the cached entity over a clone since not everybody has access to / enabled the guild members intent to receive that update to actually patch it.

Reply:
Do we want to update the cached entity with potentially unreliable user input? The API method returns a 204 Empty response on success, so we don't have any Discord-validated data to update with.